moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/current_state_monitor.h

```diff
@@ -50,7 +50,7 @@
 namespace planning_scene_monitor
 {
 
-typedef std::function<void(const sensor_msgs::msg::JointState::ConstPtr& joint_state)> JointStateUpdateCallback;
+typedef std::function<void(const sensor_msgs::msg::JointState::ConstSharedPtr& joint_state)> JointStateUpdateCallback;
```

Comment:
Why are you changing pointer to shared?

Reply:
ConstPtr is deprecated and throws warnings. As why it should be translated to ConstSharedPtr, if we have no reason to make it unique or weak, should be shared. Check slides 7-8 of https://roscon.ros.org/2015/presentations/state-of-ros2.pdf .